rust/src/lib/ip.rs

```diff
@@ -1143,3 +1171,51 @@ impl MergedInterface {
         Ok(())
     }
 }
+
+// User might define IPv6 token in the format of `::0.0.250.193`, which should
+// be sanitize to `::fac1`.
+fn sanitize_ipv6_token_to_string(
+    token: &mut String,
+) -> Result<(), NmstateError> {
+    // Empty token means reverting to default "::"
+    if token.is_empty() {
+        token.clear();
+        write!(token, "::").ok();
```

Comment:
If the string is empty it means it has a length of zero. And at the same time `clear()`  will make the string have a length of zero (empty). So I think checking if it is empty and then clearing is overkill IMO. What do you think?